cpp/src/io/utilities/file_utils.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <io/utilities/file_utils.hpp>
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <rmm/device_buffer.hpp>
+
+namespace cudf {
+namespace io {
+
+file_wrapper::file_wrapper(std::string const &filepath, int flags)
+  : fd(open(filepath.c_str(), flags))
+{
+  CUDF_EXPECTS(fd != -1, "Cannot open file");
```

Comment:
Any chance we can include the `filepath` in the error message? Would be extremely helpful if issues arise from this in distributed systems.

Reply:
Currently CUDF_EXPECTS only works with compile-time strings. Let me look into whether this can/should be changed.